config/src/lib.rs

```diff
@@ -1838,7 +1838,11 @@ fn default_initial_cols() -> u16 {
 fn default_hyperlink_rules() -> Vec<hyperlink::Rule> {
     vec![
         // URL with a protocol
-        hyperlink::Rule::new(r"\b\w+://(?:[\w.-]+)\.[a-z]{2,15}\S*\b", "$0").unwrap(),
+        hyperlink::Rule::new(
+            r"\b\w+://(?:[\w.-]+)(?:(?:\.[a-z]{2,15}\S*)|(?::\d{1,5}))\b",
```

Comment:
Looking at the regex a bit more closely, I'm not sure this is quite right; it does cover the `localhost:3000` case, but there are some cases like this that I don't think it will handle:

`http://localhost:3000/index.html`
`http://www.foo.com:3000`

(I'd suggest adding those to the tests below)

I'd suggest something like this, which I have not tested at all, which I think will allow an optional `:123` after the domain name bits, but before the `\S*` bit that is intended to match the path portion of the URL:

```
r"\b\w+://(?:[\w.-]+)(?:\.[a-z]{0,15})(?::\d{1,5})?\S*\b"
```

Reply:
It should hopefully handle the extra cases as well, plus i added the two links to the tests (hopefully correctly). However, I am not really sure if the 129 lines long array (L766 at `termwiz/src/surface/line.rs`) makes much sense.